osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/RsaEncryptionService.java

```diff
@@ -310,6 +317,36 @@ public static PrivateKey readPrivateKeyFromFile(final String filename) {
         }
     }
 
+    public static PrivateKey readPrivateKeyFromResource(final Resource resource) {
```

Comment:
A lot of copied code from readPrivateKeyFromFile. Same remark for readPublicKeyFromResource method

Reply:
I understand what you mean, although they look the same, they are initialized differently, one is String+check, other one is Resource+check, what I could do is create a generic Error Handler to remove that duplication.